### ISSUE TITLE
Confusion about the result with different training parameters

### ISSUE THREAD

**Issue by lambert-x** [ClementPinard/SfmLearner-Pytorch#79]
Hi @ClementPinard ,

Thanks for you work. I followed your instructions and have successfully done most of the work. I trained with the prepared KITTI dataset (remove the data of 2011_10_03_drive_0058 because of some .bin files are missing from the official dataset(official source, try several times) ) .  I have trained two models with the following commands:

`python3 train.py /path/to/the/formatted/data/ -b4 -m0.2 -s0.1 --epoch-size 3000 --sequence-length 3 --log-output --with-gt`
`python3 train.py /path/to/the/formatted/data/ -b4 -m0 -s2.0 --epoch-size 1000 --sequence-length 5 --log-output --with-gt`

With the first command, the validation performance reached a peak after trained with 30 epochs and got even worse with further training.

For the second case, it converged much better and could reproduce almost the same performance as your pretrained model. 

Could you explain why you set them to be 0 and 2.0 and how did you find these weights for the loss? 

By the way, did you face a problem with a very high cpu usage while training? I want to train several models together since there are some free GPU memory, but the CPU is full now.

Thanks,
Junfei

**Comment by ClementPinard** [comment]
Explaination for these hyperparameters is result driven :man_shrugging: 
The first set of hyperparameters is the first official set of hyperpamaters from the author, when he signed his paper.
The second set is based on the improved version (see [here](https://github.com/tinghuiz/SfMLearner#notes)) are a smooth loss of 0.5 applied on disparity map, and no mask loss.

unfortunately, when I recreate the exact same results, my network didn't conerge at all, so I had to improvise without diverging too much from the paper. So now I apply smooth loss on depth instead of disparity, and thus the smooth weight can be greatly increased. See here to change it back to disparity if you want : https://github.com/ClementPinard/SfmLearner-Pytorch/blob/master/train.py#L279

Here is some discussion about it : https://github.com/tinghuiz/SfMLearner/issues/124

ANd lastly, your problem with the missing .bin files is interesting, never had it. Maybe the raw dataset changed, and they removed it in order to use it at a validation somewhere else ? I'll try to add a warning to the README about this particular scene.
